src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -173,49 +178,7 @@ export class SearchSource {
   /**
    * returns all search source fields
    */
-  getFields(recurse = false): SearchSourceFields {
-    let thisFilter = this.fields.filter; // type is single value, array, or function
-    if (thisFilter) {
-      if (typeof thisFilter === 'function') {
-        thisFilter = thisFilter() || []; // type is single value or array
-      }
-
-      if (Array.isArray(thisFilter)) {
-        thisFilter = [...thisFilter];
-      } else {
-        thisFilter = [thisFilter];
-      }
-    } else {
-      thisFilter = [];
-    }
-
-    if (recurse) {
-      const parent = this.getParent();
-      if (parent) {
-        const parentFields = parent.getFields(recurse);
-
-        let parentFilter = parentFields.filter; // type is single value, array, or function
-        if (parentFilter) {
-          if (typeof parentFilter === 'function') {
-            parentFilter = parentFilter() || []; // type is single value or array
-          }
-
-          if (Array.isArray(parentFilter)) {
-            thisFilter.push(...parentFilter);
-          } else {
-            thisFilter.push(parentFilter);
-          }
-        }
-
-        // add combined filters to the fields
-        const thisFields = {
-          ...this.fields,
-          filter: thisFilter,
-        };
-
-        return { ...parentFields, ...thisFields };
-      }
-    }
+  getFields(): SearchSourceFields {
```

Comment:
If I understand it correctly, this new `getFields` will not return the `parent` field, as it is now removed in `constructor`, I guess that is intentional?

```ts
  /**
   * Returns all search source fields, except `parent` field.
   */
  getFields(): SearchSourceFields {
```

Reply:
yes, this is how the method looked in 7.11